### PR TITLE
[FIX] Added proper 2fa handling on the password reset methods

### LINKED ISSUE #10183: Password reset does not work with 2FA enabled
### Description:

If you enable two-factor authentication for your account, you will not be able to reset your password via the forgot password link.

This needs to be fixed to enable the 2FA pop-up modal check after entering a new password on the reset password link OR bypass 2FA (bad) for the reset password link.

### Steps to Reproduce:

1. Enable 2FA for your account
2. Logout and click forgot password, enter your email
3. Click the password reset link in your email
4. Enter new password
5. Get error that you cannot reset your password
6. Console error should show the error being "TOTP Required [totp-required]"

### PR DESCRIPTION
**Pull request by Hudell**
Closes #10183

This PR fixes the handling of 2FA on the password reset method

### CODE REVIEW

File: packages/rocketchat-2fa/server/methods/resetPasswordWithTOTP.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+				backupTokens: user.services.totp.hashedBackup
+			});
+
+			if (verified !== true) {
+				throw new Meteor.Error('totp-invalid', 'TOTP Invalid');
+			}
+
+			try {
+				const result = Meteor.call('resetPassword', userToken, hashedPassword);
+				console.log(result);
```
Comment (rodrigok) [inline packages/rocketchat-2fa/server/methods/resetPasswordWithTOTP.js]: Can you remove this log?

**Comment by rafaelks** [comment]
@rodrigok We're using REST today to reset the user's password... can we have a REST also?

**Comment by sampaiodiego** [comment]
@Hudell can you please solve the conflicts and fix @rodrigok pointed out issues?

**Comment by pierre-lehnen-rc** [comment]
This feature was re-implemented by another PR.

**Pull request closed without merging**